test/pylib/rest_client.py

```diff
@@ -78,3 +85,74 @@ async def put_json(self, resource: str, json: dict, host: str) \
+        if resp.status != 200:
+            text = await resp.text()
+            raise RuntimeError(f"status code: {resp.status}, body text: {text}, "
+                               f"resource {resource} params {params}")
+        return resp
+
+
+class ScyllaRESTAPIClient():
+    """Async Scylla REST API client"""
+
+    def __init__(self, port: int = 10000):
+        self.cli = TCPRESTClient(port)
+
+    async def stop(self):
+        """Close session"""
+        await self.cli.stop()
+
+    async def get_server_uuid(self, server_id: str):
+        """Get server id (UUID)"""
+        host_uuid = await self.cli.get_text("/storage_service/hostid/local", host=server_id)
+        host_uuid = host_uuid.lstrip('"').rstrip('"')
+        return host_uuid
+
+    # TODO: change to UUID
+    async def wait_for_host_known(self, dst_server_id: str, expect_server_id: str,
+                                  timeout: float = 30.0, sleep: float = 0.05) -> bool:
+        """Checks until dst_server_id knows about expect_server_id, with timeout"""
+        max_time = time() + timeout
+        while True:
+            resp_body = await self.cli.get_text("/storage_service/host_id", dst_server_id)
+            if expect_server_id in resp_body:
+                logger.info("wait_for_host_known [{%s}] found, SUCCESS", expect_server_id)
+                return True
+            if time() > max_time:
+                logger.info("wait_for_host_known [{%s}] timeout", expect_server_id)
+                return False
+            logger.debug("wait_for_host_known [{%s}] not found, sleeping {sleep}", expect_server_id)
+            await asyncio.sleep(sleep)
+
+    async def remove_node(self, initiator_ip: str, server_uuid: str) -> None:
+        """Initiate remove node of server_uuid in initiator initiator_ip"""
+        resp = await self.cli.post("/storage_service/remove_node", params={"host_id": server_uuid},
+                                   host=initiator_ip)
+        logger.debug("remove_node status %s for %s", resp.status, server_uuid)
+
+    async def wait_for_remove(self, initiator_ip: str, server_uuid, timeout: int = 30) -> None:
+        """Wait until server remove finishes"""
+        max_time = time() + timeout
+        while True:
+            resp_body = await self.cli.get_text("/storage_service/removal_status",
+                                                host=initiator_ip)
+            if not server_uuid in resp_body:
+                logger.info("wait_for_remove SUCCESS %s %s", server_uuid, resp_body)
+                break
+            if time() > max_time:
+                logger.error("wait_for_remove timeout reached %s", server_uuid)
+                break
+            logger.info("wait_for_remove WAIT %s", server_uuid)
+            await asyncio.sleep(0.05)
+
+    async def remove_node_and_wait(self, initiator_ip: str, server_uuid: str, timeout: int = 30) \
```

Comment:
Please remove the default value for `timeout` and pass it from the call site.

Reply:
No waits so no timeout in https://github.com/scylladb/scylladb/commit/7cc3aedb8d083a86de5cf4d92fd4c3190af4413d